src/uu/date/src/date.rs

```diff
@@ -136,6 +136,24 @@ impl From<&str> for Rfc3339Format {
     }
 }
 
+/// Check if we should use UTC time based on the utc flag and TZ environment variable
+fn should_use_utc(utc: bool) -> bool {
+    // Either the -u flag is set or the TZ environment variable is empty
+    utc || matches!(std::env::var("TZ"), Ok(tz) if tz.is_empty())
```

Comment:
Maybe this is _slightly_ nicer: `std::env::var("TZ").is_ok_and(|tz| tz.is_empty())`